lib/tracing-policy.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+function RateLimiterPolicy(samplesPerSecond) {
+  var sampleDelay = 1000 / samplesPerSecond;
+  this.traceWindow = sampleDelay < 1 ? 1 : sampleDelay;
+  this.lastTraceStart = Date.now() - sampleDelay;
+}
+
+RateLimiterPolicy.prototype.shouldTrace = function(dateMillis) {
+  var enable = dateMillis - this.lastTraceStart >= this.traceWindow;
```

Comment:
Keep `this.nextTraceStart` instead of `this.lastTraceStart` in the state. Reduces the number of subtracts from the hot path.